src/components/create-note-box/create-note-box.jsx

```diff
@@ -2,26 +2,59 @@ import React, { Component } from "react";
 import './create-note-box.css'
 
 export default class NoteBox extends Component {
+    
+    constructor() {
+        super();
+
+        this.state = {
+        editing: false
+        };
+
+        this.obBtnEdit = () => {
+            this.setState({
+               editing: true
+              });       
+        };
+
+        this.editingDone = event => {
+            if (event.key === "Enter") {
+            this.setState({ editing: false })
+          };
+        };
+    };
+    
     render() {
-        const { data, onBtnDelete} = this.props   
- 
-      
+        const { data, onBtnDelete, setUpdate} = this.props    
+        
+        const editMode = {};
+        editMode.display = 'none';
+
+            if (this.state.editing) {
+                editMode.display = 'block';
+            }
+
+        
         return (
             <div className='note-container'
-                 key={data.id}>
-                
-            <div>{data.id}</div>
-            <button className='delete-btn'
+                >
+                <div>{data.id}</div>
+                <button className='delete-btn'
                     onClick={(id) => onBtnDelete(data.id, id)}>
                     X
                 </button>
-            <button className='edit-btn'>Edit
+                <button className='edit-btn'
+                    onClick={this.obBtnEdit}
+                >Edit
                 </button>
-            <div>{data.title}</div>
+                <textarea className='text-area'
+                    style={editMode}
```

Comment:
this can be done through if statement, which will make it more clear. Something like that
```
{this.state.editing ? <div><div/> :  <textarea ... />}
```
React also has an empty node, which will be better than div